### ISSUE TITLE
Amend GA4 tag placement logic to use Google Tag ID if set

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#6080]
Instead of assuming that the GA4 web data stream measurement ID is the correct ID to inject in the frontend, the `Analytics_4\Web_Tag` implementation needs to instead place the Google Tag ID determined by the module (will be implemented in a separate issue) _if_ it is already set. Existing sites may not have it set yet, so in those cases it still needs to fall back to using the measurement ID like it has been doing.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `Analytics_4` module should, _if_ the `gteSupport` feature flag from #6078 is enabled and _if_ the new `googleTagID` module setting from #6077 is set, use that tag ID in the JS gtag snippet, instead of the `measurementID`.
    * It should however continue to use `measurementID` as a fallback if that value is not set / empty.

## Implementation Brief
* Using `includes/Modules/Analytics_4.php`,
	* Within the `register_tag` method,
		* Check if the `gteSupport` feature flag is enabled via the `Feature_Flags::enabled` method.
		* Check if `googleTagID` from the module settings (`$settings`) is set and if it's the case, instantiate `Web_Tag` with the `googleTagID` settings value, if not, fallback to the existing `measurementID`.

### Test Coverage
* Add tests for the above changes.

### QA:Eng

- In order to QA this, a GA4 property is required with Google Tag settings populated. However, that won't be possible until #6081 (the task that actually populates Google Tag settings) is merged.
- Wait until #6081 is merged.
- Enable the `gteSupport` feature flag from the tester plugin.
- Connect the Analytics module, with a GA4 property having Google Tag settings.
- Make sure GA4 snippet is inserted in Analytics module settings.
- View the page source of a WordPress page.
- Look for a `<script>` tag with the ID `google_gtagjs-js-after`.
- Verify that the tag placement for GA4 is the Google Tag ID, instead of the GA4 measurement ID.
- If the `gteSupport` feature flag is disabled, verify that the Google Tag ID is replaced with the fallback GA4 measurement ID.

## Changelog entry

* Use the configured Google Tag ID for the Analytics 4 snippet if present.

**Comment by techanvil** [comment]
IB :white_check_mark:

**Comment by bethanylang** [comment]
**NOTE**: This issue is blocked by #6078 in Sprint 91.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

@kuasha420 it's my understanding that we were going to do some quick checks and move this ticket to approved.  We need to make sure any additional testing would be picked up in #6081.  My understanding is that you completed some regression testing and all seems good. 

I have therefore moved this to approved!

**Comment by aaemnnosttv** [comment]
Gave this a quick review as well; a simple change that will get more testing before we start to roll it out as well, but LGTM 👍
